processor/resourcedetectionprocessor/factory.go

```diff
@@ -158,6 +158,10 @@ func (f *factory) getResourceDetectionProcessor(
 	cfg config.Processor,
 ) (*resourceDetectionProcessor, error) {
 	oCfg := cfg.(*Config)
+	err := oCfg.Validate()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This part is not needed AFAIK. `cfg.Validate()` is called from the core config validation. But still please verify it by trying this out.

Reply:
Yes, It is not needed and I have verified the same.